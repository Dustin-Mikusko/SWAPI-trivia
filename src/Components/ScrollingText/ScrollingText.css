

.fade {
  position: relative;
  width: 100%;
  /* min-height: 10vh; */
  top: 1px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.scrolling-text-container {
  display: flex;
  justify-content: center;
  position: relative;
  height: 40vh;
  overflow: hidden;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
  color: #e8c539;
}

.scrolling-text-p-font {
  color: #e8c539;
  font-family: 'Pathway Gothic One', sans-serif;
}

.movie-title {
  color: #e8c539;
  font-family: 'Pathway Gothic One', sans-serif;
}

.scrolling-text {
  color: #e8c539;
  font-family: 'Pathway Gothic One', sans-serif;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 150s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
    0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
    100% {
    top: -3500px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
