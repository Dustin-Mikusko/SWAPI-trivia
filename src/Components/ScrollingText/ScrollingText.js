import React from 'react';
import './ScrollingText.css';

const ScrollingText = (props) => {
  console.log(props);
  return (
    <div>

      <div className="fade"></div>

        <section className="scrolling-text-container">
          <div className="crawl">
            <div className="title">
              <p className="scrolling-text-p-font">Episode {props.episodeNumber}</p>
              <h1 className="movie-title">Episode {props.movieTitle}</h1>
            </div>
            <p className="scrolling-text">{props.movieText.replace("\r\n", "<br>")}</p>
            </div>
        </section>
    </div>
  )
}

export default ScrollingText;
