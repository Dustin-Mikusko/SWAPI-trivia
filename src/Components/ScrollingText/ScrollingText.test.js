import React from 'react';
import { shallow } from 'enzyme';
import ScrollingText from './ScrollingText';
