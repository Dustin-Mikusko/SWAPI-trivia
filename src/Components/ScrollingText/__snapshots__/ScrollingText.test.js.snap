// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should render component to match snapshot 1`] = `
<div>
  <div
    className="fade"
  />
  <section
    className="scrolling-text-container"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p
          className="scrolling-text-p-font"
        >
          Episode 
          4
        </p>
        <h1
          className="movie-title"
        >
          Episode 
          Revenge of the Sith
        </h1>
      </div>
      <p
        className="scrolling-text"
      >
        Hello
      </p>
    </div>
  </section>
</div>
`;
